modules/targets/generic-linux.nix

```diff
@@ -52,6 +52,24 @@ in {
       . "${pkgs.nix}/etc/profile.d/nix.sh"
       . "${profileDirectory}/etc/profile.d/hm-session-vars.sh"
     '';
+    
+    programs.zsh.envExtra = ''
+      # Make system functions available to zsh
+      () {
+        setopt LOCAL_OPTIONS CASE_GLOB EXTENDED_GLOB
+
+        local system_fpaths=(
```

Comment:
How about providing an option to add extra fpaths?

Reply:
That should probably go in the ZSH module itself, yeah? If it did, then this PR could be rewritten to use that API instead.

On a related note, I have a re-write of the ZSH module which already has this feature (and several others that I brought up awhile back [here](https://github.com/nix-community/home-manager/issues/1088#issuecomment-759895111)). I'd be happy to make a PR for that, but I intentionally dropped support for some things like prezto and OhMyZSH because (1) it's hard to keep track of what's doing what (does prezto call `compinit`? OhMyZSH? What side-effects do they have that affect the HM module?) and (2) a plugin manager isn't needed when using home-manager (my re-write is a superset of Prezto/OhMyZSH/zimfw).

It's a big breaking change and I'm new here, but if that's something the community is open to I'd happily help.